components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/services/MessageFlowReporterThread.java

```diff
@@ -267,12 +268,26 @@ void addFaultsToParents(List<StatisticsLog> messageFlowLogs, BasicStatisticDataU
             StatisticsLog updatingLog = messageFlowLogs.get(index);
             updatingLog.incrementNoOfFaults();
             index = updatingLog.getParentIndex();
+            setElasticMetaData(dataUnit, updatingLog);
+        }
+
+    }
 
-            if (dataUnit.getElasticMetadata() != null) {
-                updatingLog.setElasticMetadata(dataUnit.getElasticMetadata());
+    private void setElasticMetaData(BasicStatisticDataUnit dataUnit, StatisticsLog statisticsLog) {
+        if (dataUnit.getElasticMetadata() == null) {
+            dataUnit.generateElasticMetadata(dataUnit.getMessageContext());
```

Comment:
Why we are generating metadata here? Was that logic there before too?

Reply:
Previously what we had was,
``` java
if (dataUnit.getElasticMetadata() != null) {
	statisticsLog.setElasticMetadata(elasticMetadata);
}
```

I am in the process of changing the code so that data unit is not altered. Will update with that.

Reply:
We are passing the custom properties as an object map, within the field in the ElasticMetadata->contextProperties.
Here, if the elastic metadata in the data unit is null, we have to synthesize metadata using the message context, before putting the custom properties inside.